aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -0,0 +1,228 @@
+   * @param skipArchive if Glue should skip an old table version when creating a new version in a commit.
+   *                    By default Glue archives all old table versions after an UpdateTable call.
+   *                    but Glue has a default max number of archived table versions (can be increased).
+   *                    So for streaming use case with lots of commits, it is recommended to turn this feature off.
+   * @param tableIdentifier table identifier
+   */
+  public GlueTableOperations(
+      Configuration conf,
+      GlueClient glue,
+      String catalogId,
+      boolean skipArchive,
+      TableIdentifier tableIdentifier) {
+    this.conf = conf;
+    this.glue = glue;
+    this.catalogId = catalogId;
+    this.skipArchive = skipArchive;
+    this.databaseName = IcebergToGlueConverter.getDatabaseName(tableIdentifier);
+    this.tableName = IcebergToGlueConverter.getTableName(tableIdentifier);
+    this.fullName = String.format("glue.%s.%s.%s", catalogId, databaseName, tableName);
+  }
+
+  @Override
+  public FileIO io() {
+    if (fileIO == null) {
+      fileIO = new HadoopFileIO(conf);
+    }
+    return fileIO;
+  }
+
+  /**
+   * Returns table name in the form glue.catalogId.databaseName.tableName
+   * @return table name
+   */
+  @Override
+  protected String tableName() {
+    return fullName;
+  }
+
+  @Override
+  protected void doRefresh() {
+    String metadataLocation = null;
+    Optional<Table> tableOptional = getGlueTable();
+    if (tableOptional.isPresent()) {
+      Table table = tableOptional.get();
+      GlueToIcebergConverter.validateTable(table, tableName());
+      metadataLocation = table.parameters().get(METADATA_LOCATION_PROP);
+    } else {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchTableException("Cannot find Glue table %s after refresh, " +
+            "maybe another process deleted it or revoked your access permission", tableName());
+      }
+    }
+    refreshFromMetadataLocation(metadataLocation);
+  }
+
+  @Override
+  protected void doCommit(TableMetadata base, TableMetadata metadata) {
+    String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
+    boolean exceptionThrown = true;
+    boolean isUpdate = false;
+    Table glueTable = null;
+    try {
+      Optional<Table> glueTableOptional = getGlueTable();
+      if (glueTableOptional.isPresent()) {
+        glueTable = glueTableOptional.get();
+        // If we try to create the table but the metadata location is already set, then we had a concurrent commit
+        if (base == null && glueTable.parameters().get(METADATA_LOCATION_PROP) != null) {
+          throw new AlreadyExistsException("Cannot commit because table %s already exists in Glue", tableName());
+        }
+        isUpdate = true;
+        LOG.debug("Committing existing Glue table: {}", tableName());
+      } else {
+        LOG.debug("Committing new Glue table: {}", tableName());
```

Comment:
I think it would make more sense to move these into `persistGlueTable` since that's when the actual call is made.